src/molecules/Modal/Modal.native.js

```diff
@@ -0,0 +1,246 @@
+import React from 'react';
+import RNModal from 'react-native-modal';
+import styled from 'styled-components/native';
+import PropTypes from 'prop-types';
+import Space from '../../atoms/Space';
+import Flex from '../../atoms/Flex';
+import Size from '../../atoms/Size';
+import View from '../../atoms/View';
+import Button from '../../atoms/Button';
+
+const CenteredModalBody = styled(View)`
+  background-color: ${(props) => props.theme.colors.background[100]};
+  border-radius: ${(props) => props.theme.spacings.xsmall};
+`;
+
+const FullscreenModalBody = styled(View)`
+  background-color: ${(props) => props.theme.colors.background[400]};
+`;
+
+const BottomsheetModalBody = styled(View)`
+  background-color: ${(props) => props.theme.colors.background[200]};
+  border-top-left-radius: 8px;
+  border-top-right-radius: 8px;
+  shadow-color: rgba(11, 112, 231, 0.1); /**TODO: confirm color from theme */
+  shadow-offset: 0px -4px;
+  shadow-opacity: 1;
+  shadow-radius: 15px;
+  elevation: 2;
+`;
+
+const FullScreenModalHeader = styled(View)`
+  background-color: ${(props) => props.theme.colors.background[100]};
+  shadow-color: rgba(11, 112, 231, 0.1); /**TODO: confirm color from theme */
+  shadow-offset: 0px 4px;
+  shadow-opacity: 1;
+  shadow-radius: 15px;
+  elevation: 2;
+`;
+
+const FullScreenDragBar = styled(View)`
+  background-color: ${(props) => props.theme.colors.shade[920]};
+  border-radius: 4px;
+`;
+
+const Divider = styled(View)`
+  background-color: ${(props) => props.theme.colors.shade[920]};
+  height: 1px;
+  width: 100%;
+`;
+
+const ModalHeader = ({ children, type, onClose }) => {
+  return type === 'centered' ? (
+    <Space padding={[1.5, 2, 0.5, 2]}>
+      <View>{children}</View>
+    </Space>
+  ) : type === 'fullscreen' ? (
+    <Flex flexDirection="row" justifyContent="space-between">
+      <Space padding={[1, 2]}>
+        <FullScreenModalHeader>
+          {children}
+          <Button
+            testID="fullscreen-cross"
```

Comment:
```suggestion
            testID="close-button"
```
Why do you need different id for different types of close? at a time only one variant will be open right?